flowey/flowey_core/src/node/github_context.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Core types and traits used to read GitHub context variables.
+
+use crate::node::user_facing::GhContextVar;
+use crate::node::ClaimVar;
+use crate::node::NodeCtx;
+use crate::node::ReadVar;
+use crate::node::StepCtx;
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::BTreeMap;
+
+#[derive(Serialize, Deserialize)]
+pub struct Head {
+    #[serde(rename = "ref")]
+    pub head_ref: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GhContextVarReaderEventPullRequest {
+    pub head: Head,
+}
+
+pub mod state {
+    pub enum Root {}
+    pub enum Event {}
+}
+
+#[derive(Clone)]
+pub struct GhVarState {
+    pub raw_name: String,
+    pub backing_var: String,
+    pub is_secret: bool,
+    pub is_object: bool,
+}
+
+pub struct GhContextVarReader<'a, S> {
+    pub ctx: NodeCtx<'a>,
+    pub _state: std::marker::PhantomData<S>,
+}
+
+impl<S> GhContextVarReader<'_, S> {
+    fn read_var<T: Serialize + DeserializeOwned>(
+        &self,
+        var_name: String,
```

Comment:
this helper can be made marginally more ergonomic by using `impl AsRef<str>` here. that way - you don't need to invoke `.to_string()` or `.into()` when passing `&'static str` args to it